apps/example-orchestrator-ui/components/Metadata/Products/Products.tsx

```diff
@@ -0,0 +1,170 @@
+    dataDisplayParams,
+    setDataDisplayParam,
+}) => {
+    const hiddenColumns: Array<keyof Product> = [];
+
+    const tableColumns: TableColumns<Product> = {
+        name: {
+            field: PRODUCT_FIELD_NAME,
+            name: COLUMN_LABEL_NAME,
+            width: '110',
+        },
+        description: {
+            field: PRODUCT_FIELD_DESCRIPTION,
+            name: COLUMN_LABEL_DESCRIPTION,
+            width: '400',
+        },
+        tag: {
+            field: PRODUCT_FIELD_TAG,
+            name: COLUMN_LABEL_TAG,
+        },
+        productType: {
+            field: PRODUCT_FIELD_PRODUCT_TYPE,
+            name: COLUMN_LABEL_PRODUCT_TYPE,
+        },
+        status: {
+            field: PRODUCT_FIELD_STATUS,
+            name: COLUMN_LABEL_STATUS,
+        },
+        productBlocks: {
+            field: PRODUCT_FIELD_PRODUCT_BLOCKS,
+            name: COLUMN_LABEL_PRODUCT_BLOCKS,
+        },
+        createdAt: {
+            field: PRODUCT_FIELD_CREATED_AT,
+            name: COLUMN_LABEL_CREATED_AT,
+        },
+    };
+
+    const { data, isFetching } = useStringQueryWithGraphql<
+        ProductsResult,
+        GraphqlQueryVariables<Product>
+    >(
+        GET_PRODUCTS_GRAPHQL_QUERY,
+        {
+            first: dataDisplayParams.pageSize,
+            after: dataDisplayParams.pageIndex * dataDisplayParams.pageSize,
+            sortBy: dataDisplayParams.sortBy,
+            filterBy: dataDisplayParams.filterBy,
+        },
+        'products',
+        true,
+    );
+
+    const totalItemCount = data
+        ? parseInt(data.products.pageInfo.totalItems)
+        : 0;
+
+    const dataSorting: DataSorting<Product> = {
+        columnId: dataDisplayParams.sortBy?.field ?? PRODUCT_FIELD_NAME,
+        sortDirection: dataDisplayParams.sortBy?.order ?? SortOrder.ASC,
+    };
+
+    const pagination: Pagination = {
+        pageSize: dataDisplayParams.pageSize,
+        pageIndex: dataDisplayParams.pageIndex,
+        pageSizeOptions: DEFAULT_PAGE_SIZES,
+        totalItemCount: totalItemCount,
+    };
+
+    return (
+        <>
+            <EuiSpacer size="m" />
+            <Table
+                data={data ? mapApiResponseToSubscriptionTableData(data) : []}
+                columns={tableColumns}
+                hiddenColumns={hiddenColumns}
+                dataSorting={dataSorting}
+                onDataSort={getDataSortHandler<Product>(
+                    dataDisplayParams,
+                    setDataDisplayParam,
+                )}
+                pagination={pagination}
+                isLoading={isFetching}
+                onCriteriaChange={getPageSizeHandler<Product>(
+                    setDataDisplayParam,
+                )}
+            />
+        </>
+    );
+};
+
+function mapApiResponseToSubscriptionTableData(
+    graphqlResponse: ProductsResult,
+): Product[] {
+    return graphqlResponse.products.page.map((product): Product => {
+        const {
```

Comment:
I think this can be shortened to something like return { ... product }

Reply:
Yeah. I changed it. I'm not sure if I feel that's explicit enough though.